corehq/apps/data_interfaces/models.py

```diff
@@ -217,6 +218,16 @@ def _try_date_conversion(self, date_or_string):
 
         return date_or_string
 
+    def clean_datetime(self, timestamp):
+        if not isinstance(timestamp, datetime):
+            timestamp = datetime.combine(timestamp, time(0, 0))
+
+        if timestamp.tzinfo:
+            # Convert to UTC and make it a naive datetime for comparison to datetime.utcnow()
+            timestamp = timestamp.astimezone(pytz.utc).replace(tzinfo=None)
```

Comment:
hmm, looks like that takes a string though and at this point i've already got a timezone-aware datetime